docs/fast_data/faq/failure_handling.md

```diff
@@ -39,18 +39,37 @@ In particular, the recover mechanism works as follow:
 The detection of issues on Mia-Platform components (e.g. pod in crash-loop backoff or consumer lag too high) is done by an alarm system that comes out-of-the-box with Mia-Platform Fast Data solution. Details can be found [here](/fast_data/monitoring/overview.md). This allows the technical team owner of the Fast Data project to be notified as soon as the issue occurs.
 
-An issue on Mia-Platform components would cause a temporary misalignment between data on Single View and data on source database. However, this misalignment will be automatically recovered as soon as Mia-Platform components return online.
+An issue on Mia-Platform components would cause a temporary misalignment between data on Single View and data on the source database. However, this misalignment will be automatically recovered as soon as Mia-Platform components return online.
 
-In particular, the recover mechanism works as follow:
+In particular, the recovery mechanism works as follows:
 * Mia-Platform components stop or crash
-* Data changes continue to get accumulated on Kafka
+* Data changes continue to accumulate on Kafka
 * Mia-Platform components stop reading data changes from the topics because they are offline
 * Mia-Platform components return online
 * Mia-Platform components restart to process Kafka messages starting from the last processed message and update the Single Views
 
+In the next paragraphs, some possible failure handling on the Real-Time Updater is described.
+
+### What if Real-Time Updater fails to process a CDC message?
+The Real-Time Updater could fail to process a message from ingestion topics for different reasons, e.g. due to the presence of a malformed message on the ingestion topic or due to a violation of a unique index on the MongoDB projection.
+In such cases, the service is designed to stop processing messages as soon as it encounters an error. In particular, by configuring the `PAUSE_TOPIC_CONSUMPTION_ON_ERROR` environment variable of the service, you can choose whether to make the service crash (and, therefore, stop consuming from all the ingestion topics) or to make it stop consuming only from the topic where the error occurred.
+We do not skip messages because the order of consumption of events does matter in Fast Data. Indeed, skipping messages, and eventually processing them later on, could lead to overriding data with old values.
+
+### How to resume from a Real-Time Updater failure?
+Technically, to resume from a failure situation it is sufficient to reset the Kafka offset of the Real-Time Updater's consumer group.
+
+### How do I recognize a Real-Time Updater failure?
+In the situation of a Real-Time Updater failure, the lag of its consumer group starts growing as soon as the error occurs.
+The Fast Data solution comes with configurable out-of-the-box Alerts on the consumer group lag. With these alerts, the teams can be immediately notified of the channel of preference.
+
+### What are the impacts of a Real-Time Updater failure?
```

Comment:
@danibix95 I have some doubts about this paragraph. What do you think about it?

Reply:
Aside from K8s probes, which restart the service in case of failure with dependencies or during message processing, I am not sure we have an automatic mechanism to recover from error state.

It is possible to define metrics that notify when lag accumulates over topics and/or services restarts too many times.

Reply:
@danibix95 How would you rephrase it?